File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ns = set(nums)
        longest = 0
        for n in ns:
            if n-1 not in ns:
                l = 1
                while (n+l in ns):
                    l+=1
                longest = max(longest,l)
        return longest

File: Data Structures & Algorithms/trapping-rain-water/submission-6.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        if (height.size() == 0)
            return 0;
        int l = 0, r = height.size() - 1, ma = 0, lh = height[l], rh = height[r];
        while (l <r) {
            if (lh < rh) {
                lh = max(height[++l], lh);
                ma+= lh-height[l];
            } else {
                rh = max(height[--r], rh);
                ma+= rh-height[r];
            }
        }
        return ma;
    }
};
